runtime/include/QuantumDevice.hpp

```diff
@@ -103,6 +104,20 @@ struct QuantumDevice {
      */
     [[nodiscard]] virtual auto GetDeviceShots() const -> size_t = 0;
 
+    /**
+     * @brief Set the PRNG of the device.
+     *
+     * @param gen The std::mt19937 PRNG object.
+     */
+    virtual void SetDevicePRNG(std::mt19937 *gen){};
+
+    /**
+     * @brief Get the PRNG of the device.
+     *
+     * @return `std::mt19937*`
+     */
+    [[nodiscard]] virtual std::mt19937 *GetDevicePRNG() { return nullptr; };
```

Comment:
Tried seeing if we needed this function but I don't see any uses for it in the code (besides what can be replaced with `this->gen`). Is this just out of habit to define a getter as well?

Reply:
In the `LightningDriver` test, to test that the setter is setting correctly, we would need a getter to retrieve the device's prng. So we either implement a getter or we make the device prng public. I lean towards the former.

Reply:
Is that a useful test though? It seems very trivial.

Reply:
I mean the test right above it is serving the same purpose on shots, but I guess shots is user facing and seeding is only internal? Happy to remove this test though, if we are ok with codecov saying the setter is not covered (not that this test would make codecov say the setter is covered anyway...)

Reply:
I removed the getter and the associated trivial tests